feature/news/src/main/java/com/teamwable/news/notice/NewsNoticeRoute.kt

```diff
@@ -0,0 +1,71 @@
+package com.teamwable.news.notice
+
+import android.content.Context
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.teamwable.designsystem.theme.WableTheme
+import com.teamwable.news.NewsViewModel
+import com.teamwable.news.R
+import com.teamwable.news.model.NewsInfoModel
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+internal fun NewsNoticeRoute(
+    viewModel: NewsViewModel = hiltViewModel(),
+    navigateToDetail: (NewsInfoModel) -> Unit
+) {
+    val context = LocalContext.current
+
+    viewModel.dummyNotice.apply {
+        if (this.isNotEmpty()) {
+            NewsNoticeScreen(context = context, notices = this, navigateToDetail = navigateToDetail)
+        } else {
+            NewsNoticeEmptyScreen()
+        }
+    }
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun NewsNoticeScreen(
+    context: Context,
+    notices: List<NewsInfoModel>,
+    navigateToDetail: (NewsInfoModel) -> Unit
+) {
+    Box(modifier = Modifier.fillMaxSize()) {
+        LazyColumn(modifier = Modifier.fillMaxSize()) {
+            items(notices) { notice ->
```

Comment:
p1 : 
```suggestion
            items(items = notices, key = {item -> item.id}) {
```

이런식으로 LazyColumn에서는 key를 명시해주어야 스크롤 시에 재사용 되는 컴포저블 뷰(item)이 stable로 인식하고 리컴포지션을 유발하지 않는 것으로 알고 잇어요!

Reply:
그리고 여기서 Box와 LazyColuunm모두 fillMaxSize()가 되어있는데 fillMaxWidth()를 재사용가능한 컴포저블에 적용하게 되면 재사용 가능한 컴포저블은 부모레이아웃의 항상 전체 너비를 차지하므로 옆으로 배치한다거나, 크기를 조정 가능한 유연성을 잃을 수 잇어요!

재사용 가능한 컴포저블에서는 최상위 레벨에 크기를 강제하는 Modifier를 생략하고, 크기 관련 Modifier는 호출 시점에 외부에서 제공하는것이 낫습니다.

![image](https://github.com/user-attachments/assets/b1937ac7-6c0e-4e1e-9282-7c03a0a8e875)
![image](https://github.com/user-attachments/assets/fd99d7cf-9574-42ee-a7a3-bbc0c2ed5c76)

Reply:
배차은우 그는 이제 컴포즈 장인이 되어 버렸다 ..

Reply:
아 맞아요 34기에 들었던 기억이 있는데
전 불필요한 컴포지션이 너무 싫어요... 근데 줄이는게 너무 빡셈 ^^;
좋은 자료 감사해용 굿